### ISSUE TITLE
Calculating with checkboxes

### ISSUE THREAD

**Issue by SaschaValentin** [mavoweb/mavo#511]
I´ve noticed that calculating with checkboxes in Mavo doesn´t seem to work. If I sum two checkboxes the result is 2 - even so I gave them two values e.g. 1200 and 600. When adding three of them it´s 3 and so on. So I guess it just adds 1 for true.

I´ve even tried to set the value of the checkbox by  mv-value="1200" but that doesn´t work either. So I hope you can tell me if it is possible at all to use value of checkboxes in Mavo to calculate with it. Or is this function not provided?

Thanks

**Comment by LeaVerou** [comment]
Indeed, checkbox value is always true/false, which is treated as 1/0 in mathematical expressions. However, this is an interesting request, and would make it consistent with how radios behave as well, so we could consider it. 
If it's not too much trouble, could you describe your use case a bit?
Also, what would you expect to happen when no value attribute is set on the checkbox?

@karger, thoughts?

**Comment by SaschaValentin** [comment]
Hi Lea,
thanks for your reply and your interest in this case.
Sure, I can tell you my concrete issue: The users of the app I´m trying to realise should have the chance to choose different newspapers, represented by the checkboxes. Each newspaper has an amount of edition. This should be the value of the checkbox, so that at the end the whole edition number is shown by sum up the single items.

I´m not that big Javascript guru, but I think this could be easily done with selecting the value of the checkbox. Therefore it would be very nice to see Mavo acting this way. Especially because this way would cover both cases - if someone needs yes/no, 1/0 and also if a special value for a checkbox is needed. Just dependig on what value the developer sets. As you said: Behave the same way as radios with the difference to choose more than one item.

Would be a great feature 👍

**Comment by karger** [comment]
I've stumbled across this old issue while hunting for a different issue.  This is a problem that can be solved using Mavo's "where" clause for filtering; I've made a codepen here: https://codepen.io/karger/pen/eYZzxbo?editors=1100 .  The checkboxes work; If you edit you can also add more papers.

I do think it makes sense, if the user sets a value on the checkbox, for that to be the value of the property if the checkbox is checked (default true if no value).  Because there's no way to specify a value for *not* checked, I'm not sure how generally useful this will be, since in many cases you'll need to treat it as a boolean anyway to handle the unchecked case, but it work for this particular use case and I can see it being useful for others too.

**Comment by SaschaValentin** [comment]
Thanks a lot for your solution. It works perfectly.

**Comment by karger** [comment]
You're welcome.  We'd love to see what you've built; it helps us understand more about what Mavo needs to be a useful tool
